hack/release-tool.py

```diff
@@ -0,0 +1,162 @@
+
+
+def generate_resource_values(product, stream_name, architecture, private):
+    arch_suffix, private_suffix = "", ""
+
+    if product == 'okd':
+        return 'origin', stream_name
+
+    if architecture != 'amd64':
+        arch_suffix = f'-{architecture}'
+
+    if private:
+        private_suffix = '-priv'
+
+    return f'{product}{arch_suffix}{private_suffix}', f'{stream_name}{arch_suffix}{private_suffix}'
+
+
+def validate_server_connection(ctx):
+    with oc.options(ctx), oc.tracking(), oc.timeout(60):
+        try:
+            username = oc.whoami()
+            version = oc.get_server_version()
+            logger.debug(f'Connected to APIServer running version: {version}, as: {username}')
+        except (ValueError, OpenShiftPythonException, Exception) as e:
+            logger.error(f"Unable to verify cluster connection using context: \"{ctx['context']}\"")
+            raise e
+
+
+def create_patch(action, custom_message, custom_reason):
+    data = {
+        'image': {
+            'metadata': {
+                'annotations': {}
+            }
+        },
+        'metadata': {
+            'annotations': {}
+        },
+        'tag': {
+            'annotations': {}
+        }
+    }
+
+    if action == 'accept':
+        phase = 'Accepted'
+    elif action == 'reject':
+        phase = 'Rejected'
+    else:
+        raise ValueError(f'Unsupported action specified: {action}')
+
+    data['image']['metadata']['annotations']['release.openshift.io/phase'] = phase
+    data['metadata']['annotations']['release.openshift.io/phase'] = phase
+    data['tag']['annotations']['release.openshift.io/phase'] = phase
+
+    message = f'Manually {action}ed per TRT'
+    if custom_message is not None:
+        message = custom_message
+
+    data['image']['metadata']['annotations']['release.openshift.io/message'] = message
+    data['metadata']['annotations']['release.openshift.io/message'] = message
+    data['tag']['annotations']['release.openshift.io/message'] = message
+
+    if custom_reason is not None:
+        data['image']['metadata']['annotations']['release.openshift.io/reason'] = custom_reason
+        data['metadata']['annotations']['release.openshift.io/reason'] = custom_reason
+        data['tag']['annotations']['release.openshift.io/reason'] = custom_reason
+
+    return data
+
+
+def write_backup_file(name, release, data):
+    ts = int(round(time.time() * 1000))
+    backup_filename = f'{name}_{release}-{ts}.json'
+
+    with open(backup_filename, mode='w+', encoding='utf-8') as backup:
+        logger.debug(f'Creating backup file: {backup_filename}')
+        backup.write(json.dumps(data, indent=4))
+
+    return backup_filename
+
+
+def update(ctx, action, ns, name, release, custom_message, custom_reason, execute):
+    patch = create_patch(action, custom_message, custom_reason)
+    logger.debug(f'Generated oc patch:\n{json.dumps(patch, indent=4)}')
+
+    with oc.options(ctx), oc.tracking(), oc.timeout(5*60):
```

Comment:
That's a big timeout - why?

Reply:
Just an oversight!  Lowered to 15 seconds.